go-controller/pkg/factory/factory.go

```diff
@@ -1152,7 +1152,7 @@ func WithUpdateHandlingForObjReplace(funcs cache.ResourceEventHandler) cache.Res
 			// This occurs not so often, so log this occurance.
 			klog.Infof("Object %s/%s is replaced, invoking delete followed by add handler", newObj.GetNamespace(), newObj.GetName())
 			funcs.OnDelete(old)
-			funcs.OnAdd(new)
+			funcs.OnAdd(new, false)
```

Comment:
Ah. That's an interesting undocumented API change... I see that this `isInInitialList` was introduced here: https://github.com/kubernetes/client-go/commit/5d70a118df80fd8ddb8fbd041af95acae49ae617

It's the end of the day for me and I don't understand how it works... :-/  Did you get what the logic behind it is? Here  for instance we're passing `isInInitialList=true` in `AddFunc` and `isInInitialList=false` in `UpdateFunc`.

Reply:
what I gathered is that using onAdd on an obj that is already exitisting we use `isInInitialList=true` otherwise
if it's something new, like in below snippet, then `isInInitialList=false`:

```
UpdateFunc: func(old, new interface{}) {
			oldObj := old.(metav1.Object)
			// This occurs not so often, so log this occurance.
			klog.Infof("Object %s/%s is replaced, invoking delete followed by add handler", newObj.GetNamespace(), newObj.GetName())
			funcs.OnDelete(old)
			funcs.OnAdd(new, false)
```